src/jace/translator/util/jace_translation_memento.py

```diff
@@ -0,0 +1,76 @@
+# JaCe - JAX Just-In-Time compilation using DaCe (Data Centric Parallel Programming)
+#
+# Copyright (c) 2024, ETH Zurich
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+from collections.abc import Mapping, Sequence
+from dataclasses import dataclass
+from typing import Any
+
+import dace
+
+
+@dataclass(init=True, repr=True, eq=False, frozen=True, kw_only=True, slots=True)
+class JaCeTranslationMemento:
```

Comment:
What about something straightforward like `SDFGContext`, `TranslationContext` or something along these lines? It would be shorter, simpler to remember and easier to understand. `TranslatedSDFG` is slightly misleading in my opinion since this class contains more stuff than a single SDFG, but does not inherit or extend the SDFG class in any other way.

Reply:
I agree that `TranslatedSDFG` is misleading, I have now settled with `TranslatedJaxprSDFG`.
I do not use `*Context` as I decided to not use this class inside the driver, instead I created a different class.
The main motivation reason for this is that in my view both are exceptionally rather different.